tests/monster_test_generated.rs

```diff
@@ -176,78 +177,45 @@ pub mod example {
   extern crate flatbuffers;
   use self::flatbuffers::EndianScalar;
 
+bitflags::bitflags! {
 /// Composite components of Monster color.
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct Color(pub u8);
-#[allow(non_upper_case_globals)]
-impl Color {
-  pub const ENUM_MIN: u8 = 1;
-  pub const ENUM_MAX: u8 = 8;
-  pub const Red: Self = Self(1);
-  /// \brief color Green
-  /// Green is bit_flag with value (1u << 1)
-  pub const Green: Self = Self(2);
-  /// \brief color Blue (1u << 3)
-  pub const Blue: Self = Self(8);
-  pub const ENUM_VALUES: &'static [Self] = &[
-    Self::Red,
-    Self::Green,
-    Self::Blue,
-  ];
-  /// Returns the variant's name or "" if unknown.
-  pub fn variant_name(self) -> &'static str {
-    match self {
-      Self::Red => "Red",
-      Self::Green => "Green",
-      Self::Blue => "Blue",
-      _ => "",
-    }
-  }
-}
-impl std::fmt::Debug for Color {
-  fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-    let name = self.variant_name();
-    if name.is_empty() {
-      f.write_fmt(format_args!("<UNKNOWN {:?}>", self.0))
-    } else {
-      f.write_str(name)
-    }
+  pub struct Color: u8 {
+    const RED = 1;
+    const GREEN = 2;
+    const BLUE = 8;
   }
 }
+
 impl<'a> flatbuffers::Follow<'a> for Color {
   type Inner = Self;
   #[inline]
   fn follow(buf: &'a [u8], loc: usize) -> Self::Inner {
-    Self(flatbuffers::read_scalar_at::<u8>(buf, loc))
+    let bits = flatbuffers::read_scalar_at::<u8>(buf, loc);
+    unsafe { Self::from_bits_unchecked(bits) }
   }
 }
 
 impl flatbuffers::Push for Color {
     type Output = Color;
     #[inline]
     fn push(&self, dst: &mut [u8], _rest: &[u8]) {
-        flatbuffers::emplace_scalar::<u8>(dst, self.0);
+        flatbuffers::emplace_scalar::<u8>(dst, self.bits());
     }
 }
 
 impl flatbuffers::EndianScalar for Color {
   #[inline]
   fn to_little_endian(self) -> Self {
-    Self(u8::to_le(self.0))
+    let bits = u8::to_le(self.bits());
+    unsafe { Self::from_bits_unchecked(bits) }
   }
   #[inline]
   fn from_little_endian(self) -> Self {
-    Self(u8::from_le(self.0))
+    let bits = u8::from_le(self.bits());
+    unsafe { Self::from_bits_unchecked(bits) }
```

Comment:
Okay, this is UB again. Welp, not real UB, because we don't use invalid values, and bitflags are integers, not enums, thus technically there are no UB. But this is at the very least unsound, because it allows any user to get an invalid value for bitflags in safe code.

Reply:
bitflags/bitflags/issues/200 documents that this `unsafe` means a crate-semantics mistake, rather than a memory / UB mistake. Their semantics are "only known flags may be set" whereas that's just not a promise Flatbuffers makes. I almost want to upstream an option to generate code with "unchecked bits are cool" semantics and methods.

Reply:
Welp, it is unsafe method, and I can't see any other way to create invalid value for bitflags, thus somebody somewhere may rely on that invariant. Though, there's no place in the generated bitflags code, where breaking "only-set-of-defined-values" invariant is relied upon for any purpose, so it is a very minor unsoundness.

Btw, why does `Scalar` trait exist? Can't we just flip the endianness in `Push` implementation itself for the types that need such flipping?

Reply:
Let's resume this conversation after bitflags/bitflags#228